src/foundry/foundry.js/hooks.d.ts

```diff
@@ -1,830 +1,804 @@
+   *   // [...]
+   * })
+   * ```
+   *
+   * @example Using a callback with a dynamic name
+   * ```typescript
+   * Hooks.on<Hooks.GetCompendiumDirectoryEntryContext>('getJournalEntryContext', (jq, entryOptions) => {
+   *   // [...]
+   * })
+   * ```
+   *
+   * @example Using a callback with a dynamic name and generic parameter
+   * ```typescript
+   * Hooks.on<Hooks.CloseApplication<FormApplication>>('closeFormApplication', (app, jq) => {
+   *   // [...]
+   * })
+   * ```
+   */
+  namespace Hooks {
+    interface StaticCallbacks {
+      /**
+       * @remarks Called when applying custom active effect changes.
+       * @param actor  - the Actor to whom this effect should be applied
+       * @param change - the change data being applied
+       * @remarks This is called by {@link Hooks.call}.
+       * @see {@link ActiveEffect#_applyCustom}
+       */
+      applyActiveEffect: (actor: Actor, change: EffectChangeData) => boolean | void;
+
+      /**
+       * @remarks This is called before a {@link Canvas} is drawn.
+       * @param canvas - the Canvas
+       * @remarks This is called by {@link Hooks.callAll}.
+       * @see {@link Canvas#draw}
+       */
+      canvasInit: (canvas: Canvas) => unknown;
+
+      /**
+       * @remarks This is called when a {@link Canvas} is panned. When called during animated panning, the callback is
+       * called on every tick.
+       * @param canvas - the Canvas
+       * @param view   - the CanvasView
+       * @remarks This is called by {@link Hooks.callAll}.
+       * @see {@link Canvas#pan}
+       * @see {@link Canvas#animatePan}
+       */
+      canvasPan: (canvas: Canvas, view: Canvas.View) => unknown;
+
+      /**
+       * @remarks This is called after a {@link Canvas} is done initializing.
+       * @param canvas - the Canvas
+       * @remarks This is called by {@link Hooks.call}.
+       * @see {@link Canvas#draw}
+       */
+      canvasReady: (canvas: Canvas) => boolean | void;
+
+      /**
+       * A hook event that fires when the Sidebar tab is changed.
+       * @param app - The SidebarTab application which is now active
+       * @remarks This is called by {@link Hooks.callAll}.
+       * @see {@link Sidebar#_onChangeTab}
+       */
+      changeSidebarTab: (app: SidebarTab) => unknown;
+
+      /**
+       * @remarks This is called when creating a {@link ChatBubble}, but before displaying it.
+       * @param token   - the speaking token
+       * @param jq      - the JQuery for the chat bubble
+       * @param message - the spoken message text
+       * @param options - additional options
+       * @param emote   - whether to style the speech bubble as an emote
+       * @remarks This is called by {@link Hooks.call}.
+       * @remarks An explicit return value of `false` prevents the chat bubble being shown.
+       * @see {@link ChatBubbles#say}
+       */
+      chatBubble: (
+        token: Token,
+        jq: JQuery,
```

Comment:
That's not a parameter, that's just the variable name that was passed. I prefer to give it a more meaningful name if it is already passed a JQuery object instead of an HTMLElement.

Reply:
But actually thinking more about it, the user can name it what they want anyway, since it's a lambda.